server/lib/nl/detection/context.py

```diff
@@ -100,13 +99,15 @@ def merge_with_context(uttr: nl_uttr.Utterance, default_place: Place = None):
   main_vars, cmp_vars = _detect_vars(
       uttr, query_type == nl_uttr.QueryType.CORRELATION_ACROSS_VARS)
 
-  # 6. Detect entities leveraging context
-  entities = _detect_entities(uttr)
-
-  # 7. Detect properties leveraging context
+  # 6. Detect properties leveraging context
   properties = _detect_props(uttr)
 
-  # 6. Populate the returned dict
+  # 7. Detect entities leveraging context. Should detect entities after
+  #    properties because we only want to use context if properties didn't use
+  #    context.
```

Comment:
sorry, don't follow.  What's an example for why entities should happen after props?

Reply:
if properties use context, we dont' want entities to also use context because then we would jsut be returning the exact same result as the previous query even though the user asked for something else